virttest/storage.py

```diff
@@ -336,8 +336,12 @@ def image_secret_define_by_params(cls, image, params):
     def save_to_file(self):
         """Save secret data to file."""
         _make_secret_dir()
-        with open(self.filename, "w") as fd:
-            fd.write(self.data)
+        try:
+            with open(self.filename, "w") as fd:
+                fd.write(self.data)
+        except TypeError:
+            with open(self.filename, "wb") as fd:
```

Comment:
do we need this? this file likes a cfg file, created by VT, so we might not need to consider the binary file.

Reply:
Yes, we do. For non utf8 string, we have to store it as binary file. And then create object secrect by \"--object secret,id={s.aid},file={s.filename}\". Thanks.